src/main/java/org/gridsuite/study/server/InsensitiveStringToEnumConverterFactory.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright 2002-2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.gridsuite.study.server;
+
+import lombok.Data;
+import org.springframework.core.convert.converter.Converter;
+import org.springframework.core.convert.converter.ConverterFactory;
+import org.springframework.lang.NonNull;
+import org.springframework.lang.Nullable;
+import org.springframework.stereotype.Component;
+import org.springframework.util.Assert;
+
+import java.util.EnumSet;
+import java.util.Locale;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+/**
+ * Converts from a String to a {@link Enum} by calling {@link Enum#valueOf(Class, String)},
+ * while being case-insensitive.
+ *
+ * @implNote based on existing spring converters code
+ *
+ * @see org.springframework.core.convert.support.StringToEnumConverterFactory
+ * @see org.springframework.boot.convert.LenientStringToEnumConverterFactory
+ */
+@Component //spring-boot implicit WebMvcConfigurer#addFormatters(registry.addConverter(...))
+@SuppressWarnings({"rawtypes", "unchecked"})
+public final class InsensitiveStringToEnumConverterFactory implements ConverterFactory<String, Enum> {
```

Comment:
This class is just to convert a string into an enum while being case insensitive ?
There is easier way to do it no ?
https://stackoverflow.com/questions/24157817/jackson-databind-enum-case-insensitive
https://vianneyfaivre.com/tech/spring-boot-enum-as-parameter-ignore-case
It wouldn't be enough ?

Reply:
`spring.jackson.mapper.ACCEPT_CASE_INSENSITIVE_ENUMS = true `
?

Reply:
I tried this but haven't worked...  
It seem that Spring don't use Jackson API to deserialize here but its own API (a `ConverterFactory`).

Reply:
Indeed it does not, well OK...